week-8/Homework/mandatory/2-fetch-exercise/exercise.js

```diff
@@ -1,26 +1,8 @@
-/*
-Use fetch to load a greeting from the API and display it 
-in the HTML element with the id "greeting-text".
 
-API: https://codeyourfuture.herokuapp.com/api/greetings
-Response: A greeting in a random language
-
-To learn more about fetch, refer to the doc:
-https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
-
-
-================
-Expected result
-================
-
-Open index.html in your browser. Every time you refresh the page,
-a different greeting should be displayed in the box.
-*/
-
-fetch('*** Write the API address here ***')
-    .then(function(response) {
-        return response.text();
-    })
-    .then(function(greeting) {
-        // Write the code to display the greeting text here
-    });
\ No newline at end of file
+fetch ('https://codeyourfuture.herokuapp.com/api/greetings')
+  .then (function (response) {
+    return response.text ();
+  })
+  .then (function (greeting) {
+    document.querySelector ('#greeting-text').innerHTML = greeting;
```

Comment:
This works, but we should generally be wary of injecting HTML into our page which comes from sources we don't trust. I'd think about setting `textContent` rather than `innerHTML`, so that the browser doesn't try to interpret the `greeting` as HTML :)

Reply:
Thanks, Daniel I have updated all the recommendations now.